src/group_theory/order_of_element.lean

```diff
@@ -368,6 +372,30 @@ calc (univ.filter (λ a : α, a ^ n = 1)).card ≤ (gpowers (g ^ (fintype.card 
     exact le_of_dvd hn0 (gcd_dvd_left _ _)
   end
 
+section
+
+variables [group α]
+
+lemma is_cyclic.gpowers_eq_univ (ha : ∀ x : α, x ∈ gpowers a) :
```

Comment:
This isn't really about cyclic groups. `gpowers a` could be replaced with any set and the lemma still holds. Same with `is_cyclic.gpowers_to_finset_eq_univ`

Reply:
Good point :smile: